### PR TITLE
cli: make check subcommand more tolerant to failures

### PR DESCRIPTION
**Pull request by devimc**

### CODE REVIEW

File: cli/kata-check.go [hunk 1]
```diff
@@ -134,17 +134,33 @@ func getCPUFlags(cpuinfo string) string {
...
 	path := filepath.Join(sysModuleDir, module)
 	if katautils.FileExists(path) {
 		return true
 	}
 
+	// Kernel module can be built-in, check /lib/modules/x.x.x before
+	// trying to load the module
+	cmd := exec.Command(modProbeCmd, "-R", module)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		kmodLog.WithError(err).Warnf("Module can not be inserted: %s. Is it built-in?", string(output))
```
Comment (amshinde) [inline cli/kata-check.go:149]: Maybe we should actually check if the module is built-in here by taking a look at /sys/modules.

Comment (devimc) [reply cli/kata-check.go:149]: we are already looking for it in above chuck of code, hence this code may not be required, thanks

**Comment by devimc** [comment]
@amshinde change applied, thanks

File: cli/kata-check.go [hunk 2]
```diff
@@ -134,17 +134,25 @@ func getCPUFlags(cpuinfo string) string {
...
+
 	// First, check to see if the module is already loaded
 	path := filepath.Join(sysModuleDir, module)
 	if katautils.FileExists(path) {
 		return true
 	}
 
+	// Only root can load modules
+	if os.Getuid() != 0 {
+		kmodLog.Warn("Module is not loaded and it can not be inserted. Please consider running with sudo or as a root")
```
Comment (jodh-intel) [inline cli/kata-check.go:147]: - Nit: s/or as a root/or as root/.
- Should this be `.Error()`?

File: cli/kata-check.go [hunk 3]
```diff
@@ -134,17 +134,25 @@ func getCPUFlags(cpuinfo string) string {
...
+		return false
+	}
+
 	// Now, check if the module is unloaded, but available.
 	// And modprobe it if so.
 	cmd := exec.Command(modProbeCmd, module)
 	if output, err := cmd.CombinedOutput(); err != nil {
-		kataLog.WithField("module", module).WithError(err).Warnf("modprobe insert module failed: %s", string(output))
+		kmodLog.WithError(err).Warnf("modprobe insert module failed: %s", string(output))
```
Comment (jodh-intel) [inline cli/kata-check.go:155]: You could keep this structured using something like:

```suggestion
		kmodLog.WithError(err).WithField("output", string(output)).Warn("modprobe insert module failed")
```

File: cli/release.go [hunk 4]
```diff
@@ -276,12 +277,17 @@ func getReleases(releaseURL string, includeAll bool) ([]semver.Version, map[stri
 
 	releasesArray := []map[string]interface{}{}
 
-	bytes, err := ioutil.ReadAll(resp.Body)
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to read release details: %v", err)
+	} else if resp.StatusCode == http.StatusForbidden && bytes.Contains(body, []byte("limit exceeded")) {
+		// Do not fail if rate limit is exceeded
```
Comment (jodh-intel) [inline cli/release.go:284]: Good catch! Hopefully unlikely for users to hit this (current unauthenticated limit seems to be 60 requests / hour).

File: cli/release.go [hunk 5]
```diff
@@ -276,12 +277,17 @@ func getReleases(releaseURL string, includeAll bool) ([]semver.Version, map[stri
...
+		kataLog.WithField("url", releaseURL).
+			Warn("API rate limit exceeded. Try again later. Read https://developer.github.com/v3/#rate-limiting for more information")
```
Comment (jodh-intel) [inline cli/release.go:286]: I'm not 100% happy about hard-coding this URL (including the "v3", but given we want to be as helpful to users as possible, I guess we'll just have to maintain this (or hope GH auto-redirect if they ever remove "v3" entirely).

**Review by jodh-intel — APPROVED** [review]
Thanks @devimc! A bunch of nits, so...

lgtm

**Comment by devimc** [comment]
@jodh-intel changes applied, thanks

/test

**Review by GabyCT — APPROVED** [review]

**Pull request merged**